src/app/pages/stats-page/payout-search/payout-search.store.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { EMPTY, Observable } from 'rxjs';
+import { catchError, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { Farmer } from 'src/app/interfaces/farmer.interface';
+import { Payout, PayoutAddress } from 'src/app/interfaces/payout.interface';
+import { FarmerService } from 'src/app/services/api/farmer.service';
+import { PayoutService } from 'src/app/services/api/payout.service';
+
+export interface PayoutSearchState {
+  isLoading: boolean;
+  error: string;
+  results: Payout | null;
+  addresses: PayoutAddress | null;
+  farmer: Farmer | null;
+}
+
+const initialState: PayoutSearchState = {
+  isLoading: false,
+  error: null,
+  results: null,
+  addresses: null,
+  farmer: null
+};
+
+@Injectable({ providedIn: 'root' })
+export class PayoutSearchStore extends ComponentStore<PayoutSearchState> {
+  constructor(private payoutService: PayoutService, private farmerService: FarmerService) {
+    super(initialState);
+  }
+
+  readonly searchPayout = this.effect((queries$: Observable<number>) => {
+    return queries$.pipe(
+      tap(() => {
+        this.setState({
+          ...initialState,
+          isLoading: true,
+        });
+      }),
+      switchMap((query: number) => {
+        return this.payoutService
```

Comment:
Don't know if this query is necessary. GET /payout/:id doesn't seem to return anything new that wouldn't be already available on the page. You could be passing the whole Payout object in effect if necessary.
On the other point if the plan is to make this component reusable elsewhere, this could be a good approach.
Up to you.